### PR TITLE
Prevents error message "no TicketID" on sending an answer if no TicketID exists.

### LINKED ISSUE #126: Bug - Regression Error Message in Log
### Environment

- OS:           macOS
- Browser:      Safari
- Znuny version: 6.0.36 & 6.1.1.

### Expected behaviour

If I send a Email reply and no error exists, no error should be written the log.

### Actual behaviour

An error is written every time to the log. 

```
ERROR: OTRS-CGI-89 Perl: 5.30.0 OS: linux Time: Mon Sep 6 13:41:39 2021

 Message: No TicketID is given!

 RemoteAddress: 172.16.39.81
 RequestURI: /otrs/index.pl?

 Traceback (175900):
   Module: Kernel::Output::HTML::Layout::Error Line: 1032
   Module: Kernel::Output::HTML::Layout::ErrorScreen Line: 1013
   Module: Kernel::Modules::AgentTicketCompose::Run Line: 62
   Module: Kernel::System::Web::InterfaceAgent::Run Line: 1136
   Module: ModPerl::ROOT::ModPerl::Registry::opt_otrs_bin_cgi_2dbin_index_2epl::handler Line: 39
   Module: (eval) (v1.99) Line: 207
   Module: ModPerl::RegistryCooker::run (v1.99) Line: 207
   Module: ModPerl::RegistryCooker::default_handler (v1.99) Line: 173
   Module: ModPerl::Registry::handler (v1.99) Line: 32
```

### How to reproduce

1. Create a ticket
2. Reply using a templates
3. Send mail
4. Error is written to the log

### Additional information

The issue was introduced in:

https://github.com/znuny/Znuny/pull/88 proposed by @reneeb 
I think this line is causing the issue:
https://github.com/znuny/Znuny/pull/88/commits/8eb11ab71968d26327bf3f1457ae28d306ff603a#diff-343bf31c8bb9d19c878033dc908936a96ca9213109401adf40731381c141d4f3R49

it only checks on change, which is made every time right before the submit is executed, because the input fields are set to ro. 
The check fails because right at the time of the execution, the popup is already closed. 

Possible solutions (proposal):
1) Perform check for needed attributes before executing the AJAX call
2) Modify the event trigger to listen for content/value change, not a change of the field itself

### LINKED ISSUE #170: "No TicketID is given!" on each outgoing reply
While working on switching from OTRS 6.0.30 to Znuny 6.0.38, I noticed the following issue in Znuny.

### Environment
- OS: CentOS Linux release 7.9.2009 (Core)
- Browser: Chrome 95.0.4638.69 (Official Build) (64-bit)
- Znuny version: 6.0.38

### Expected behaviour
Agent's ticket replies should not generate in otrs.log the error
```
[Error][Kernel::Output::HTML::Layout::Error][1030] No TicketID is given!
```

### Actual behaviour
Every agent's reply on an existing ticket generates the `No TicketID` error line following the `ArticleSend` log message.
```
[Fri Nov  5 19:44:43 2021][Info][Kernel::System::Ticket::Article::Backend::Email::ArticleSend] Queued email to '"XXX" <XXX>' from 'xxx'. HistoryType => SendAnswer, Subject => Re: [Ticket#0143090] eee;
[Fri Nov  5 19:44:44 2021][Error][Kernel::Output::HTML::Layout::Error][1030] No TicketID is given!
```

### How to reproduce
Click reply on an existing ticket, select a template, the click "Send mail", watch otrs.log.

### Additional information
* The message appears only for existing tickets. For a new email outgoing ticket generated by an agent, it does not appear.
* The issue is not present in OTRS 6.0.30

### PR DESCRIPTION
**Pull request by reneeb**
## Proposed change

- '1 - 🐞 bug 🐞'            - Bugfix (non-breaking change which fixes an issue)

## Additional information

In some situations when an agent sends an answer the CheckSubject function is called, but the TicketID isn't passed.

- This PR is related to forum discussion: https://community.znuny.org/viewtopic.php?f=35&t=42655

## Checklist

- [X] The code change is tested and works locally.(❗)
- [X] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [ ] Local ZnunyCodePolicy run passes successfully.(❕)
- [ ] Local unit tests pass.(❕)
- [x] GitHub workflow ZnunyCodePolicy passes.(❗)
- [x] GitHub workflow unit tests pass.(❗)

(Replacement PR for #118)

### CODE REVIEW

File: var/httpd/htdocs/js/Core.Agent.TicketCompose.js [hunk 1]
```diff
@@ -75,16 +75,22 @@ Core.Agent.TicketCompose = (function (TargetNS) {
 
     function CheckSubject () {
         var CurrentSubject = $('#Subject').val();
+        var ForTicketID    = $('input[name=TicketID]').val();
```
Comment (dennykorsukewitz) [inline var/httpd/htdocs/js/Core.Agent.TicketCompose.js]:
```suggestion
        var TicketID    = $('input[name=TicketID]').val();
```

File: var/httpd/htdocs/js/Core.Agent.TicketCompose.js [hunk 2]
```diff
@@ -75,16 +75,22 @@ Core.Agent.TicketCompose = (function (TargetNS) {
...
+
         $('#SubjectWarning').remove();
 
+        if ( !ForTicketID ) {
```
Comment (dennykorsukewitz) [inline var/httpd/htdocs/js/Core.Agent.TicketCompose.js]:
```suggestion
        if ( !TicketID ) {
```

File: var/httpd/htdocs/js/Core.Agent.TicketCompose.js [hunk 3]
```diff
@@ -75,16 +75,22 @@ Core.Agent.TicketCompose = (function (TargetNS) {
...
+
         $.ajax({
             url: Core.Config.Get('Baselink'),
             type: 'POST',
             data: {
                 Action: Core.Config.Get('Action'),
                 Subaction: 'CheckSubject',
                 Subject: CurrentSubject,
-                TicketID: $('input[name=TicketID]').val(),
+                TicketID: ForTicketID,
```
Comment (dennykorsukewitz) [inline var/httpd/htdocs/js/Core.Agent.TicketCompose.js]:
```suggestion
                TicketID: TicketID,
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]

File: var/httpd/htdocs/js/Core.Agent.TicketCompose.js (hunk 3 above)
Comment (reneeb) [reply var/httpd/htdocs/js/Core.Agent.TicketCompose.js]: Changed...

**Review by dennykorsukewitz — APPROVED** [review]
Thank you! :rocket:

**Pull request merged**